### PR TITLE
circleci: Split out sanitizer tests from core2. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Also, allow EM_FROZEN_CACHE=0 to work as expected (rather than setting
it to a string which is truthy).

Split out from #17659.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -209,6 +209,10 @@ commands:
           # further blocked by issue #6897.
           name: run tests
           environment:
+            # There are a tests in the browser test suite that using library
```
Comment (kripken) [inline .circleci/config.yml:212]:
```suggestion
            # There are tests in the browser test suite that use libraries
```

**Review by kripken — APPROVED** [review]
lgtm, but how big is the cost of adding a new CI job? This new job is pretty short I think, would be good to make sure it's long enough to justify a new job.

**Comment by sbc100** [comment]
> lgtm, but how big is the cost of adding a new CI job? This new job is pretty short I think, would be good to make sure it's long enough to justify a new job.

The sanitizer tests need to build a bunch of extra C libraries (especially once I add more of them in #17659)..  by adding them to thier own builder like this we can actually do more stuff in parallel since we don't need to add those extra libs to the base linux build.

I'll get some numbers for the overhead of the new job and the actually runtime of the tests in the new job..

**Comment by kripken** [comment]
Hmm, we have some other asan tests in the main test suite, e.g. `core2.test_asan_no_error`. So don't we build libc for asan always anyhow in all core jobs?

**Comment by sbc100** [comment]
> Hmm, we have some other asan tests in the main test suite, e.g. `core2.test_asan_no_error`. So don't we build libc for asan always anyhow in all core jobs?

Yes, but #17659 also adds the `-mt` asan libc

**Comment by sbc100** [comment]
Looks like the new run takes 1m 59s our of which 40 are actual tests, but I imagine that #17659 will make the tests dominate (although maybe not by much).   Anyway 1.19 seems like a small enough overhead, and it runs in parallel with other tests suites

**Comment by kripken** [comment]
It does run in parallel, but if it doesn't reduce the overall work, we'd be using more power and spending more - what's the benefit in that case?

**Comment by sbc100** [comment]
> It does run in parallel, but if it doesn't reduce the overall work, we'd be using more power and spending more - what's the benefit in that case?

It allows us to build these extra libraries on demand rather than building them up front (i.e. adding them to MINIMAL). 

Alternatively we would run the whole `core2` suite with `EM_FROZEN_CACHE=0`.. but that seems non-ideal.   Its also non-ideal to keep adding more tests from other test suites to `core2`.

Overall this should speed up the overall test run.. at the cost of about 1 minute of compute (the time it takes to spin up a new runner on circleci).

**Comment by sbc100** [comment]
I split out the EM_FROZEN_CACHE stuff into https://github.com/emscripten-core/emscripten/pull/17671 so it can land independently.

**Comment by kripken** [comment]
> It allows us to build these extra libraries on demand rather than building them up front (i.e. adding them to MINIMAL).

Sorry, I'm missing something. I don't see changes to embuilder here, so MINIMAL is the same as it was before?

Removing stuff from MINIMAL might be useful for build times, I can see that, but as MINIMAL isn't changed I don't see the overall benefit here yet.

**Comment by sbc100** [comment]
The change to MINIMAL was required by #17659.   Without this change I would need to add more `asan` pic variants to MINIMAL.  This change allows the asan/lsan tests to build whatever libraries they need on demand rather than having them be part of MININAL.

**Comment by sbc100** [comment]
> The change to MINIMAL was required by #17659. Without this change I would need to add more `asan` pic variants to MINIMAL. This change allows the asan/lsan tests to build whatever libraries they need on demand rather than having them be part of MININAL.

Another way of putting it.. this change allows more asan testing in the future without extended MINIMAL.

**Review by kripken — APPROVED** [review]
I see, thanks, that's what I was missing, so this will be helpful when adding new asan stuff in the future, which will only be needed in this new CI mode.

**Comment by sbc100** [comment]
I actually found a way to avoid landing this change I think...

**Pull request closed without merging**